File: frontend/src/types/assignmentGroupInvitation.ts
import type { ClassMemberResponse } from './class';

// ============ ENUMS ============
export enum AssignmentGroupInvitationStatus {
  Pending = 0,
  Accepted = 1,
  Rejected = 2,
}

// ============ ENTITIES ============
export interface AssignmentGroupInvitation {
  id: number;
  fromMember: ClassMemberResponse;
  toMember: ClassMemberResponse;
  status: AssignmentGroupInvitationStatus;
  sentAt: Date;
  respondedAt?: Date | null;
}

// ============ REQUEST TYPES ============
export interface CreateAssignmentGroupInvitationRequest {
    fromMemberId: number;
    toMemberId: number;
}

// ============ RESPONSE TYPES ============
export interface AssignmentGroupInvitationResponse {
  id: number;
  fromMember: ClassMemberResponse;
  toMember: ClassMemberResponse;
  status: AssignmentGroupInvitationStatus;
  sentAt: Date;
  respondedAt?: Date | null;
}

File: backend/Models/Entities/AssignmentGroupInvitation.cs
using OnlineClassroomManagement.Helper.Constants;

namespace OnlineClassroomManagement.Models.Entities
{
    public class AssignmentGroupInvitation
    {
        public int Id { get; set; }
        public ClassMember FromMember { get; set; }
        public ClassMember ToMember { get; set; }
        public AssignmentGroupInvitationStatus Status { get; set; }
        public DateTime SentAt { get; set; }
        public DateTime? RespondedAt { get; set; }
    }
}